Updated import wrapper and IO module; added specman to IO

thcasey3 opened the pull request:
Updated importing and made example, unittests, and GUI compatible

ylin00 commented on data/topspin/hydrationGUI Results/E_ksig.csv:
```diff
@@ -0,0 +1,22 @@
+E powers,E(p),ksigma(p),ksigma(p) fit
```
Could you change `/hydrationGUI Results/` to `hydrationGUI_Results/`, to avoid potential issues with space in the path.

thcasey3 replied:
Done

ylin00 commented on dnplab/dnpIO/specman.py:
```diff
@@ -0,0 +1,126 @@
+        file_name_exp = path
+    elif path[-1] == "1":
+        file_name_exp = path.replace("d01", "exp")
+        file_name_d01 = path
```
mising `else`

thcasey3 replied:
done

ylin00 commented on dnplab/dnpIO/specman.py:
```diff
@@ -0,0 +1,126 @@
+    exp_file_opened = open(path, encoding="utf8", errors="ignore")
+    file_contents = exp_file_opened.read().splitlines()
+    exp_file_opened.close()
+    params = {}
```
need initializing `c`

Maybe `c = ""`

thcasey3 replied:
done

ylin00 commented on dnplab/dnpIO/specman.py:
```diff
@@ -0,0 +1,126 @@
+            params[c + "_" + splt_exp_content[0]] = splt_exp_content[1]
+        elif len(splt_exp_content) == 1 and exp_content != "":
+            params[c + "_" + str(i)] = splt_exp_content
+
```
```suggestion
            params[c + "_" + str(i)] = splt_exp_content
        else:
            pass

```

thcasey3 replied:
done

ylin00 commented on dnplab/dnpIO/specman.py:
```diff
@@ -0,0 +1,126 @@
+    uint_read = np.fromfile(file_opened, dtype=np.uint32)
+    file_opened.close()
+
+    file_opened = open(path, "rb")
```
Do you need close and reopen here?

thcasey3 replied:
If I don't reopen the file it appears that the file_opened variable is changed by np.fromfile, the second call to np.fromfile doesn't find what it needs from file_opened. The only way I can get it to work is to open it twice. Maybe there is a better way to do it but I haven't found it yet.

ylin00 replied:
I see. I think it is good enough.

ylin00 commented on dnplab/dnpIO/specman.py:
```diff
@@ -0,0 +1,126 @@
+    elif uint_read[2] == 4:
+        y_data = np.reshape(
+            float_data_folded, (uint_read[6], uint_read[5], uint_read[4], uint_read[3])
+        )
```
`else: y_data = ?`

thcasey3 replied:
i'll limit it to 4D and just raise an error message if more than 4

ylin00 commented on dnplab/dnpIO/topspin.py:
```diff
@@ -119,21 +119,20 @@ def find_group_delay(decim, dspfvs):
 
 
-def load_title(path, expNum=1, titlePath="pdata/1", titleFilename="title"):
+def load_title(path=1, titlePath="pdata/1", titleFilename="title"):
```
```suggestion
def load_title(path="1", titlePath="pdata/1", titleFilename="title"):
```

ylin00 commented on dnplab/dnpIO/topspin.py:
```diff
@@ -142,20 +141,19 @@ def load_title(path, expNum=1, titlePath="pdata/1", titleFilename="title"):
 
 
-def load_acqu(path, expNum=1, paramFilename="acqus"):
+def load_acqu(path=1, paramFilename="acqus"):
```
```suggestion
def load_acqu(path="1", paramFilename="acqus"):
```

ylin00 commented on dnplab/dnpIO/topspin.py:
```diff
@@ -177,8 +175,8 @@ def load_acqu(path, expNum=1, paramFilename="acqus"):
 
-def load_proc(path, expNum=1, procNum=1, paramFilename="procs"):
-    pathFilename = path + str(expNum) + "/pdata/" + str(procNum) + "/" + paramFilename
+def load_proc(path=1, procNum=1, paramFilename="procs"):
```
```suggestion
def load_proc(path="1", procNum=1, paramFilename="procs"):
```

ylin00 commented on dnplab/dnpIO/topspin.py:
```diff
@@ -386,18 +381,17 @@ def topspin_jcamp_dx(path):
         numpy.ndarray: vdlist as numpy array
     """
-    fullPath = path + str(expNum) + "/vdlist"
+    fullPath = path + "vdlist"
```
do you need `os.path.join` ?

thcasey3 replied:
I fixed this in many more places as well

ylin00 commented on dnplab/dnpIO/topspin.py:
```diff
@@ -524,25 +516,28 @@ def topspin_ser_phase_cycle(path, expNum, paramFilename="acqus"):
             pass
 
     return dataDict
+"""
```
Delete the lines instead of comment it out.

thcasey3 replied:
done

ylin00 commented on unittests/test_dnpImport.py:
```diff
@@ -11,100 +12,35 @@ def setUp(self):
-        self.assertEqual(
-            "serPhaseCycle", topspin.dir_data_type(self.testdata, expNum=18)
-        )
-        self.assertEqual(
```
👍

ylin00 commented on unittests/test_dnpImport.py:
```diff
@@ -124,7 +60,7 @@ def setUp(self):
     def test_import_prospa_exp_is_1d(self):
         datas = [
-            prospa.import_prospa(self.test_data + "/%i/data.csv" % expNum)
+            wrapper.load(self.test_data + "/%i/data.csv" % expNum, data_type="prospa")
```
```suggestion
            wrapper.load(os.path.join(self.test_data, "%i" % expNum, "data.csv"), data_type="prospa")
```

thcasey3 replied:
done

ylin00 commented on dnplab/dnpIO/specman.py:
```diff
@@ -0,0 +1,126 @@
+    Returns:
+        specman_data (object) : dnpdata object containing specman data
+    """
+    if path[-1] == os.sep:
```
This is potentially error-prone. In windows `os.sep == \` but `path[-1]` can still be `/` 
```suggestion
    path = os.path.normpath(path)
```
Need tests

thcasey3 replied:
I added normpath to the wrapper function so the path is normalized before it gets to any of the IO functions

ylin00 commented on dnplab/dnpIO/specman.py:
```diff
@@ -0,0 +1,136 @@
+        file_name_d01 = path
+    else:
+        raise TypeError("Incorrect file type, must be .d01 or .exp")
+        return
```
no need to return after raise Exception.
```suggestion
```

ylin00 commented on dnplab/dnpIO/specman.py:
```diff
@@ -0,0 +1,136 @@
+        )
+    else:
+        raise TypeError("DNPLab currently only supports up to 4D data")
+        return
```
```suggestion
```

ylin00 commented on dnplab/hydrationGUI.py:
```diff
@@ -167,19 +167,30 @@ def __init__(self):
         self.optwidthCheckbox.resize(100, 20)
```
I didn't review changes to this script. Please let me know if you have specific things that I can help to check.

thcasey3 replied:
Ok, I think everything is in order. All tests pass, the example runs, and the GUI works fine

thcasey3 committed:
updates related to baseline and autophase functions

-added baseline function to dnpNMR
-improved autophase in dnpNMR and removed it from dnpData
-added unittests for new or improved functions
-fixed vnmrj where "t1" was called "array", fixed corresponding unittest

ylin00 commented on dnplab/dnpNMR.py:
```diff
@@ -279,6 +279,108 @@ def integrate(all_data, dim="t2", integrate_center=0, integrate_width=100):
+            )
+
+    else:
+        raise TypeError
```
```suggestion
        raise TypeError("type must be either 'poly' or 'expon'")
```

thcasey3 replied:
done

ylin00 commented on dnplab/dnpNMR.py:
```diff
@@ -279,6 +279,108 @@ def integrate(all_data, dim="t2", integrate_center=0, integrate_width=100):
+        dim (str) : dimension to correct along
+        type (str) : type of baseline fit, either "poly" for polynomial or "expon" for exponential
+        order (int) : polynomial order, or 1=mono and 2=bi for exponential
+
```
```suggestion
```

thcasey3 replied:
done

ylin00 commented on dnplab/dnpNMR.py:
```diff
@@ -279,6 +279,108 @@ def integrate(all_data, dim="t2", integrate_center=0, integrate_width=100):
+    if reference_slice is not None:
+        if len(_np.shape(data.values)) == 1:
+            reference_slice = None
+            print("ignoring reference_slice, this is 1D data")
```
use `warnings.warn` instead of print.

thcasey3 replied:
done

thcasey3 committed:
allow arbitrary s_max

-added edit box for s_max to hydrationGUI
-allow input of arbitrary s_max as long as it is numeric and between 0 and 1

thcasey3 committed:
Update dnpNMR.py

minor changes requested by ylin

ylin00 reviewed (changes requested):
This PR contains multiple edits. Have to be processed after approval of #49 .

thcasey3 committed:
Update __init__.py

thcasey3 committed:
Merge branch 'develop' into tom-devimport
